packages/react-router/docs/guides/testing.md

```diff
@@ -158,43 +99,51 @@ const App = () => (
     />
   </div>
 );
+```
 
-// the actual test!
-it("navigates around", done => {
-  renderTestSequence({
-    // tell it the subject you're testing
-    subject: App,
-
-    // and the steps to execute each time the location changes
-    steps: [
-      // initial render
-      ({ history, div }) => {
-        // assert the screen says what we think it should
-        console.assert(div.innerHTML.match(/Welcome/));
-
-        // now we can imperatively navigate as the test
-        history.push("/dashboard");
-      },
-
-      // second render from new location
-      ({ div }) => {
-        console.assert(div.innerHTML.match(/Dashboard/));
-
-        // or we can simulate clicks on Links instead of
-        // using history.push
-        Simulate.click(div.querySelector("#click-me"), {
-          button: 0
-        });
-      },
-
-      // final render
-      ({ location }) => {
-        console.assert(location.pathname === "/");
-        // you'll want something like `done()` so your test
-        // fails if you never make it here.
-        done();
-      }
-    ]
+```jsx
+// you can also use a renderer like "@testing-library/react" or "enzyme/mount" here
+import { render, unmountComponentAtNode } from "react-dom";
+import { act } from 'react-dom/test-utils';
+import { Router } from "react-router";
+import { createMemoryHistory } from "history";
+
+// app.test.js
+it("navigates home when you click the logo", async => {
+  // in a real test a renderer like "@testing-library/react"
+  // would take care of setting up the DOM elements
+  const root = document.createElement('div');
+  document.body.appendChild(root);
+  
+  const history = createMemoryHistory();
+  
+  // Set initial location
+  history.push('/my/initial/route');
+  
+  // Render app
+  render(
+    <Router history={history}>
+      <App />
+    <Router>,
+    root
+  );
+  
+  // Interact with page
+  act(() => {
+    // Find the link (perhaps using the text content)
+    const goHomeLink = document.querySelector('#nav-logo-home');
+    // Click it
+    goHomeLink.dispatchEvent(new MouseEvent("click", { bubbles: true }));
   });
+  
+  // Assert about new location
+  expect(history.location.pathname).toBe('/homepage');
```

Comment:
Instead of making an assertion on `history.location.pathname`, I'd rather see an assertion that some different `<Route>` is rendered in response to the navigation. So the test description would be like "when the home link is clicked, the home page renders". This is a much more direct assertion on the effect the router has on the page instead of the side effect it has on the URL.

Reply:
100% agree 👍

Reply:
Do you just want to move this to after the line below that does that?

Reply:
URLs are part of the API of the app, so sometimes you do need to check them as part of the test, in addition to the content.